esphome/components/modbus_controller/modbus_controller.h

```diff
@@ -458,6 +462,13 @@ class ModbusController : public PollingComponent, public modbus::ModbusDevice {
   bool get_module_offline() { return module_offline_; }
   /// Set callback for commands
   void add_on_command_sent_callback(std::function<void(int, int)> &&callback);
+  /// called by esphome generated code to set the max_cmd_retries.
+  void set_max_cmd_retries(uint8_t max_cmd_retries) {
+    // We add 1 so the initial transmission doesn't count when checking against command->send_counter.
+    this->max_cmd_retries_ = max_cmd_retries + 1;
```

Comment:
Only problem here is that `get_max_cmd_retries()` will return a different number to what was set.
```suggestion
    this->max_cmd_retries_ = max_cmd_retries;
```